src/services/doppler-api-client.ts

```diff
@@ -0,0 +1,78 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession, DopplerAPIConnectionData } from './app-session';
+import { RefObject } from 'react';
+import { SubscriberList, SubscriberListState } from './shopify-client';
+
+export interface DopplerAPIClient {
+  getListData(idList: number, apikey: string): Promise<ResultWithoutExpectedErrors<SubscriberList>>;
+}
+
+export class HttpDopplerAPIClient implements DopplerAPIClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerAPIConnectionData(): DopplerAPIConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler API connection data is not available');
+    }
+    return { jwtToken: connectionData.jwtToken, userAccount: connectionData.userData.user.email };
+  }
+
+  private mapList(data: any): SubscriberList {
+    return {
+      name: data.name,
+      id: data.listId,
+      amountSubscribers: data.subscribersCount,
```

Comment:
Why not?

```suggestion
      subscribersCount: data.subscribersCount,
```

Reply:
I'm not so sure, amount and count is debatable...

Reply:
Right, but at least it is the same than before :P